src/mat.rs

```diff
@@ -1356,6 +1356,40 @@ macro_rules! mat4s {
                 (*self * point.into_homogeneous_point()).normalized_homogeneous_point().truncated()
             }
 
+            /// If self represents an `Isometry3` (i.e. self is a product of the from `T * R` where
```

Comment:
This should work as long as self represents an Affine transformation, not just if it's a direct isometry (an affine transform is nonlinear scale, skew, rotation, and translation.